lib/symboldatabase.cpp

```diff
@@ -6398,6 +6398,16 @@ void SymbolDatabase::setValueType(Token* tok, const ValueType& valuetype, Source
                 } else if (parsedecl(vt2->containerTypeToken, &autovt, mDefaultSignedness, mSettings, mIsCpp)) {
                     setType = true;
                     templateArgType = vt2->containerTypeToken->type();
+                    if (Token::simpleMatch(autoToken->next(), "&"))
+                        autovt.reference = Reference::LValue;
+                    else if (Token::simpleMatch(autoToken->next(), "&&"))
+                        autovt.reference = Reference::RValue;
+                    if (autoToken->previous()->str() == "const") {
+                        if (autovt.pointer && autovt.reference != Reference::None)
```

Comment:
I am not sure if it affects the constness if it's a reference. Maybe I am confused is this for some code like:

     std::vector<std::string*> data;
     for (const auto& x: data) 

?

Reply:
Yes, that results in ´std::string* const&´